samples/nrf9160/tcp_client/src/main.c

```diff
@@ -0,0 +1,231 @@
+	printk("IP address %s, port number %d\n",
+	       CONFIG_TCP_SERVER_ADDRESS_STATIC,
+	       CONFIG_TCP_SERVER_PORT);
+
+	err = send(client_fd, buffer, sizeof(buffer), 0);
+	if (err < 0) {
+		printk("Failed to transmit TCP packet, %d\n", errno);
+		close(client_fd);
+		return;
+	}
+
+	k_delayed_work_submit(
+			&server_transmission_work,
+			K_SECONDS(CONFIG_TCP_DATA_UPLOAD_FREQUENCY_SECONDS));
+}
+
+static void work_init(void)
+{
+	k_delayed_work_init(&server_transmission_work,
+			    server_transmission_work_fn);
+}
+
+#if defined(CONFIG_BSD_LIBRARY)
+static void lte_handler(const struct lte_lc_evt *const evt)
+{
+	switch (evt->type) {
+	case LTE_LC_EVT_NW_REG_STATUS:
+		if ((evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_HOME) &&
+		     (evt->nw_reg_status != LTE_LC_NW_REG_REGISTERED_ROAMING)) {
+			break;
+		}
+
+		printk("Network registration status: %s\n",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming\n");
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_PSM_UPDATE:
+		printk("PSM parameter update: TAU: %d, Active time: %d\n",
+			evt->psm_cfg.tau, evt->psm_cfg.active_time);
+		break;
+	case LTE_LC_EVT_EDRX_UPDATE: {
+		char log_buf[60];
+		ssize_t len;
+
+		len = snprintf(log_buf, sizeof(log_buf),
+			       "eDRX parameter update: eDRX: %f, PTW: %f\n",
+			       evt->edrx_cfg.edrx, evt->edrx_cfg.ptw);
+		if (len > 0) {
+			printk("%s\n", log_buf);
+		}
+		break;
+	}
+	case LTE_LC_EVT_RRC_UPDATE:
+		printk("RRC mode: %s\n",
+			evt->rrc_mode == LTE_LC_RRC_MODE_CONNECTED ?
+			"Connected" : "Idle\n");
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		printk("LTE cell changed: Cell ID: %d, Tracking area: %d\n",
+		       evt->cell.id, evt->cell.tac);
+		break;
+	default:
+		break;
+	}
+}
+
+static int configure_low_power(void)
+{
+	int err;
+
+if (IS_ENABLED(CONFIG_TCP_PSM_ENABLE)) {
+	/** Power Saving Mode */
+	err = lte_lc_psm_req(true);
+	if (err) {
+		printk("lte_lc_psm_req, error: %d\n", err);
+	}
+} else {
+	err = lte_lc_psm_req(false);
+	if (err) {
+		printk("lte_lc_psm_req, error: %d\n", err);
+	}
+}
+
+if (IS_ENABLED(CONFIG_TCP_EDRX_ENABLE)) {
+	/** enhanced Discontinuous Reception */
+	err = lte_lc_edrx_req(true);
+	if (err) {
+		printk("lte_lc_edrx_req, error: %d\n", err);
+	}
+} else {
+	err = lte_lc_edrx_req(false);
+	if (err) {
+		printk("lte_lc_edrx_req, error: %d\n", err);
+	}
+}
+
+if (IS_ENABLED(CONFIG_TCP_RAI_ENABLE)) {
+	/** Release Assistance Indication  */
+	err = lte_lc_rai_req(true);
+	if (err) {
+		printk("lte_lc_rai_req, error: %d\n", err);
+	}
+}
+
+	return err;
+}
+
+static void modem_configure(void)
+{
+	int err;
+
+	if (IS_ENABLED(CONFIG_LTE_AUTO_INIT_AND_CONNECT)) {
+		/* Do nothing, modem is already configured and LTE connected. */
+	} else {
+		err = lte_lc_init_and_connect_async(lte_handler);
+		if (err) {
+			printk("Modem configuration, error: %d\n", err);
+			return;
+		}
+	}
+}
+#endif
```

Comment:
A comment here would be nice to indicate what the `#endif` corresponds to. For instance `#endif /* defined(CONFIG_LTE_LINK_CONTROL) */`